deps/rabbit/src/rabbit_variable_queue.erl

```diff
@@ -2613,14 +2614,33 @@ maybe_deltas_to_betas(DelsAndAcksFun,
                         ram_bytes            = RamBytes,
                         disk_read_count      = DiskReadCount,
                         delta_transient_bytes = DeltaTransientBytes,
-                        transient_threshold  = TransientThreshold }) ->
+                        transient_threshold  = TransientThreshold,
+                        rates                = #rates{out = RatesOut }}) ->
     #delta { start_seq_id = DeltaSeqId,
              count        = DeltaCount,
              transient    = Transient,
              end_seq_id   = DeltaSeqIdEnd } = Delta,
+    %% We limit the number of messages we read from the index. We want a large
+    %% enough value but not one that will use too much memory because when there
+    %% are many queues we may end up exploding the memory. We can use the current
+    %% rate to determine the upper value unless configured otherwise. If the rate
+    %% is too low, the min value will be used instead, and it defaults to 100
+    %% as that's neither too low nor too high.
```

Comment:
Does the 100 value result in different behaviour if message payloads vary? For example, how does the lazy queue behave for 1KB messages? What about 1MB?

Reply:
Only the index is read into memory.